File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesTransactionExceptionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.PurchasesErrorCode;
import com.revenuecat.purchases.PurchasesTransactionException;

final class PurchasesTransactionExceptionAPI {
    static void check(final PurchasesTransactionException exception) {
        final String underlyingErrorMessage = exception.getUnderlyingErrorMessage();
        final String message = exception.getMessage();
        final PurchasesErrorCode code = exception.getCode();
        final PurchasesError error = exception.getError();
        final boolean userCancelled = exception.getUserCancelled();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesTransactionExceptionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.PurchasesErrorCode
import com.revenuecat.purchases.PurchasesTransactionException

class PurchasesTransactionExceptionAPI {
    fun check(exception: PurchasesTransactionException) {
        val underlyingErrorMessage: String? = exception.underlyingErrorMessage
        val message: String = exception.message
        val code: PurchasesErrorCode = exception.code
        val error: PurchasesError = exception.error
        val userCancelled: Boolean = exception.userCancelled
    }
}
